lib/metasploit/framework/login_scanner/opnsense.rb

```diff
@@ -0,0 +1,141 @@
+require 'metasploit/framework/login_scanner/http'
+
+module Metasploit
+  module Framework
+    module LoginScanner
+
+      # This is the LoginScanner class for dealing with Deciso B.V. OPNSense instances.
+      # It is responsible for taking a single target, and a list of credentials
+      # and attempting them. It then saves the results.
+      class OPNSense < HTTP
+
+        def get_cookie_value(response, wanted_cookie_name)
+          response.get_cookies.split('; ').find { |cookie| cookie.start_with?(wanted_cookie_name) }.split('=').last
+        end
+
+        # Sends a HTTP request with Rex
+        #
+        # @param (see Rex::Proto::Http::Request#request_raw)
+        # @return [Rex::Proto::Http::Response] The HTTP response
+        def send_request(opts, keep_cookies = false)
+          res = super(opts)
+
+          if keep_cookies && res
+            @php_sessid = get_cookie_value(res, 'PHPSESSID')
```

Comment:
I'm assuming login scanners can run safely in parallel; it that assumption holds true - is it safe to assign this state as a side effect in your module? 👀